java/compaction/compaction-job-execution/src/test/java/sleeper/compaction/job/execution/CompactionJobMessageHandlerTest.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.compaction.job.execution;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import sleeper.compaction.job.CompactionJob;
+import sleeper.compaction.job.execution.CompactionJobMessageHandler.JobAndMessage;
+import sleeper.compaction.job.execution.CompactionJobMessageHandler.MessageConsumer;
+import sleeper.compaction.job.execution.CompactionJobMessageHandler.Result;
+import sleeper.configuration.properties.instance.InstanceProperties;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Queue;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static sleeper.configuration.properties.InstancePropertiesTestHelper.createTestInstanceProperties;
+import static sleeper.configuration.properties.instance.CompactionProperty.COMPACTION_TASK_MAX_CONSECUTIVE_FAILURES;
+import static sleeper.configuration.properties.instance.CompactionProperty.COMPACTION_TASK_MAX_TIME_IN_SECONDS;
+
+public class CompactionJobMessageHandlerTest {
+
+    private final InstanceProperties instanceProperties = createInstance();
+    private static final Queue<CompactionJob> JOBS_ON_QUEUE = new LinkedList<>();
+    private static final List<CompactionJob> SUCCESSFUL_JOBS = new ArrayList<>();
+    private static final List<CompactionJob> FAILED_JOBS = new ArrayList<>();
```

Comment:
It looks like these don't need to be static, and then we also wouldn't need the setup method.